src/EditorFeatures/CSharpTest/UseSimpleUsingStatement/UseSimpleUsingStatementTests.cs

```diff
@@ -0,0 +1,535 @@
+    {
+        {|FixAllInDocument:|}using (var a = b)
+        using (var c = d)
+        {
+            using (var e = f)
+            using (var g = h)
+            {
+                Console.WriteLine(a);
+            }
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        using var a = b;
+        using var c = d;
+        using var e = f;
+        using var g = h;
+        Console.WriteLine(a);
+    }
+}",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSimpleUsingStatement)]
+        public async Task TestFixAll4()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        {|FixAllInDocument:|}using (var a = b)
+        using (var c = d)
+        {
+            using (e)
+            using (f)
+            {
+                Console.WriteLine(a);
+            }
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        using var a = b;
+        using var c = d;
+        using (e)
+        using (f)
+        {
+            Console.WriteLine(a);
+        }
+    }
+}",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSimpleUsingStatement)]
+        public async Task TestWithFollowingReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        [||]using (var a = b)
+        {
+        }
+        return;
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        using var a = b;
+        return;
+    }
+}",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseSimpleUsingStatement)]
+        public async Task TestWithFollowingBreak()
+        {
+            await TestInRegularAndScriptAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        switch (0)
+        {
+            case 0:
+                {
+                    [||]using (var a = b)
+                    {
+                    }
+                    break;
+                }
+        }
+    }
+}",
+@"using System;
+
+class C
+{
+    void M()
+    {
+        switch (0)
+        {
+            case 0:
+                {
+                    using var a = b;
+                    break;
+                }
+        }
+    }
+}",
+parseOptions: CSharp8ParseOptions);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeLocalFunctionStatic)]
+        public async Task TestMissingInSwitchSection()
+        {
+            await TestMissingAsync(
+@"using System;
+
+class C
+{
+    void M()
+    {
+        switch (0)
+        {
+            case 0:
+                [||]using (var a = b)
+                {
+                }
+                break;
+        }
+    }
+}", parameters: new TestParameters(parseOptions: CSharp8ParseOptions));
+        }
+    }
+}
```

Comment:
@chsienki Did you update the spec for `using` declarations with rules for jumps? 
Would some of the restrictions on `goto` affect this analyzer/fixer?

Reply:
Yikes!  Thanks for letting me know.  I will update accordingly.

> You might consider not offering the refactoring if the block containing the using statement, or any of its child statements , are a goto or label.

Yup.  That's what i'll do.  Thanks!

Reply:
Yes, that would be fine, as there isn't any 'crossing' the using statement. It gets tricky to track them all, but you could definitely do a simple check for cases like that and then disable if you can't easily prove it's safe.

Reply:
@chsienki I went with the more conservative approach as i didn't think it was worth investing a lot of time to make these rarer cases work.  We can always revisit that later.  But, by default, we'll err on the side of caution.

Reply:
Agree, I think that's the right approach (plus goto's + using are hopefully rare enough that users will rarely come across it anyway).